### DISCUSSION TITLE
How to convert multiple schemas into a single shcema?

### DISCUSSION THREAD

**Discussion by CavalloneChen** [foxglove/community#898] [Q&A]
My custom messages come with multiple schemas and I need to combine them into a single foxglove.Grid schema. But the exaple message converter can only handle a single schema pair conversion. How can I achieve my goal?

**Comment by defunctzombie** [comment]
Message converters only support 1:1 conversions.

It sounds like you might want to look at [User scripts](https://docs.foxglove.dev/docs/visualization/panels/user-scripts/) which will let you read from multiple topics, track some state, and output to a new topic for your Grid.
